src/mercury_engine_data_structures/formats/bmsld.py

```diff
@@ -220,5 +284,53 @@ def compare_func(first: str, second: str) -> bool:
 
     def insert_into_entity_group(self, sub_area: Container, name_to_add: str) -> None:
         # MSR requires to have the names in the sub area list sorted by their crc32 value
-        sub_area.names.append(name_to_add)
-        sub_area.names.sort(key=crc32)
+        sub_area.objects.append(name_to_add)
+        sub_area.objects.sort(key=crc32)
+
+    def get_layer(self, layer: ActorLayer) -> Container:
+        """Returns a layer of actors"""
+        return self.raw.actor_layers[layer]
+
+    def _check_if_actor_exists(self, layer: ActorLayer, actor_name: str) -> None:
+        if actor_name not in self.get_layer(layer):
+            raise KeyError(f"No actor named '{actor_name}' found in '{layer}!'")
+
+    def get_actor(self, layer: ActorLayer, actor_name: str) -> Container:
+        """Returns an actor given a layer and an actor name"""
+        self._check_if_actor_exists(layer, actor_name)
+        return BmsldActor(self.raw.actor_layers[layer][actor_name])
+
+    def remove_actor(self, layer: ActorLayer, actor_name: str) -> None:
+        """Deletes an actor given a layer and an actor name"""
+        self._check_if_actor_exists(layer, actor_name)
+        self.get_layer(layer).pop(actor_name)
+        self.remove_actor_from_all_groups(actor_name)
+
+    def copy_actor(
+        self,
+        coords: list[float],
+        template_actor: Container,
+        new_name: str,
+        layer: ActorLayer,
+        offset: tuple = (0, 0, 0),
+    ) -> Container:
+        """
+        Copies an actor to a new position
+
+        param coords: the x, y, z position for the copied actor
+        param template_actor: the actor being copied
+        param new_name: the name for the copied actor
+        param layer: the layer the copied actor will be added to
+        param offset: adds an additional offset the copied actor's coordinates
+
+        """
+        new_actor = copy.deepcopy(template_actor)
+        self.raw.actor_layers[layer][new_name] = new_actor
+        for i in range(2):
+            new_actor["position"][i] = coords[i] + offset[i]
```

Comment:
`new_actor.position = Vec3([p + o for p, o in zip(new_actor.position.raw, offset)])`

Reply:
would it make sense to still use `coords` over `new_actor.position.raw`?

Reply:
yes sorry